DCA/adapter.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from albumentations import Normalize, Compose
+import ever
+import torch
+import cv2
+
+from pathlib import Path
+
+from adapter_base import AdapterBase
+from DCA.utils.my_tools import pre_slide
+from DCA.module.Encoder import Deeplabv2
+
+
+class DcaAdapter(AdapterBase):
+    
+    def __init__(self, image_size=1024, factor=2, mode="Urban"):
+        """
+        Image Size -- size to scale image to (algorithm requires square images)
+        Factor -- algorithm runs segmentation on square patches of image, factor shows how to scale each side of image to make these patches (if factor=n, there will be n^2 patches)
+        """
+
+        self._image_size = image_size
+        self._factor = factor
+        self._mode = mode
+
+    @property
+    def image_size(self):
+        return self._image_size
+    
+    @image_size.setter
+    def image_size(self, image_size):
+        self._image_size = image_size
+
+    @property
+    def factor(self):
+        return self._factor
+    
+    @factor.setter
+    def factor(self, factor):
+        self._factor = factor
+
+    @property
+    def mode(self):
+        return self._mode
+    
+    @mode.setter
+    def mode(self, mode):
+        self._mode = mode
+
+    def process(self, image: np.ndarray):
+        image = self._transform_image(image)
+        model = self._build_model()
+        raw_predictions = self._process(model, image)
+        predictions = self._postprocess_predictions(raw_predictions)
+        return predictions
+
+    
+    def _transform_image(self, image: np.ndarray):
+        """Input: MxMx3 image"""
+        image = cv2.resize(image, (self.image_size, self.image_size), interpolation=cv2.INTER_AREA)
+        print(image.shape)
+        transformer = Compose([
+            Normalize(mean=(73.53223948, 80.01710095, 74.59297778),
+                    std=(41.5113661, 35.66528876, 33.75830885),
+                    max_pixel_value=1, always_apply=True),
+            ever.preprocess.albu.ToTensor()
+        ], is_check_shapes=False)
+        blob = transformer(image=image)
+        image = blob['image'].to(torch.device('cuda' if torch.cuda.is_available() else 'cpu'))
+        image = image[None, : ]
+        return image
+
+    def _build_model(self):
+        model = Deeplabv2(dict(
+            backbone=dict(
+                resnet_type='resnet50',
+                output_stride=16,
+                pretrained=True,
+            ),
+            multi_layer=True,
+            cascade=False,
+            use_ppm=True,
+            ppm=dict(
+                num_classes=7,
+                use_aux=False,
+                fc_dim=2048,
+            ),  
+            inchannels=2048,
+            num_classes=7
+        )).to('cuda' if torch.cuda.is_available() else 'cpu')
+        ckpt_path = {"urban" : Path(__file__).parent / "weights" / "URBAN_0.4635.pth", 
```

Comment:
Why is the checkpoint name hardcoded? Why can't you just pass the path to it in the constructor?